### PR TITLE
Fix ATTACH query with external ON CLUSTER

### LINKED ISSUE #55009: On Cluster alter commands are checking for local db
When I run an alter command  with `on cluster` clause for a remote cluster, it gives me the error
```
[2023-09-26 11:47:24] Code: 81. DB::Exception: Database shard does not exist. (UNKNOWN_DATABASE) (version 23.8.2.7 (official build))
```
In the previous version (v22.8), this worked well, even when the current server didn't have the db. In the latest version (v23.8.2.7), this is giving an error.

**How to reproduce**

* Which ClickHouse server version to use: v23.8.2.7
* Lets say we have two servers: server 1 and server 2, same version.
* In server 1, Add a remote cluster (`test_cluster`) to the config (server 2 is in the remote cluster), where server 1 is not a part of the cluster. 
* Create a database (eg `test_db`) on the server 2. Make sure the same db is not in server 1.
* On server 1, run `CREATE TABLE test_db.test_table on cluster 'test_cluster' (id Int64) Engine=MergeTree;`
* On server 1, run `ALTER TABLE test_db.test_table on cluster 'test_cluster' add column id_test Int64;`

**Expected behavior**

The second command should run and add column in table on test_cluster

**Error message and/or stacktrace**
```
[2023-09-26 11:53:35] Code: 81. DB::Exception: Database shard does not exist. (UNKNOWN_DATABASE) (version 23.8.2.7 (official build))
```

### PR DESCRIPTION
**Pull request by evillique**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the ATTACH query with the ON CLUSTER clause when the database does not exist on the initiator node. Closes https://github.com/ClickHouse/ClickHouse/issues/55009

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit a0044f3aa4d5c297229be6f6829f8d7b5e7d75c5 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61365/a0044f3aa4d5c297229be6f6829f8d7b5e7d75c5/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/InterpreterCreateQuery.cpp [hunk 1]
```diff
@@ -1087,8 +1087,8 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
     // If this is a stub ATTACH query, read the query definition from the database
     if (create.attach && !create.storage && !create.columns_list)
     {
-        auto database = DatabaseCatalog::instance().getDatabase(database_name);
-        if (database->shouldReplicateQuery(getContext(), query_ptr))
+        auto database = DatabaseCatalog::instance().tryGetDatabase(database_name);
```
Comment (alexey-milovidov) [inline src/Interpreters/InterpreterCreateQuery.cpp:1090]: Add a comment with reasoning (you can copy-paste everything from PR description).

**Review by alexey-milovidov — APPROVED** [review]
Please merge with master.

**Comment by clickgapai** [comment]
Hi @evillique @alexey-milovidov — while reviewing this PR I found the following:

- **Coverage gap:** Stub ATTACH on missing database without `ON CLUSTER` — explicit `UNKNOWN_DATABASE` throw untested — https://github.com/ClickHouse/ClickHouse/pull/103925

Could you review the test PR and add the `can be tested` label if it looks good?
Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**